R/layout.R

```diff
@@ -290,291 +287,93 @@ layout_columns <- function(
-          #    rather than causing an empty row.
-          if (widths[idx] > (n_cols - cursor)) {
-            cursor <- 0L
-          }
-          row_remaining <- 0L
-        }
-        add_start_class <- row_remaining >= widths[idx] || cursor > 0
-      }
-
-      if (cursor > 0 && cursor + widths[idx] > n_cols) {
-        # adding the current item would overflow the row, so we need a start class
-        add_start_class <- TRUE
-        cursor <- 0L
-      }
-
-      if (add_start_class) {
-        add_class(idx, sprintf("g-start-%s-%s", break_name, cursor + 1L))
-        add_start_class <- FALSE
-      }
-
-      add_class(idx, sprintf("g-col-%s-%s", break_name, this_width))
-      update_cursor(this_width, is_empty = FALSE)
-    }
-  }
-
-  vapply(classes, paste, character(1), collapse = " ")
-}
-
-
-col_width_best_fit <- function(n_items, prefer_wider = FALSE, n_cols = NA) {
-  # If we're here, the user asked us to make the best choice possible about the
-  # number of columns, either by giving `col_widths = NA` or by using `NA` at
-  # a specific break point. The general idea is play with both the column
-  # widths (col_widths) and number of columns in the grid (n_cols) to
-  # get decent results for a low number of items (1-7). At 7+ we use the
-  # 12-column grid and pick the factor that results in the fewest empty columns.
-
-  fit <- list(n_cols = n_cols)
-
-  # If n_cols is NA and n <= 7, best fit can adjust the n_cols for a better fit
-  if (is.na(fit$n_cols)) {
-    fit$n_cols <- if (n_items > 7) {
-      12
-    } else if (n_items > 3) {
-      n_items * 2
-    } else {
-      n_items
-    }
-
-    if (n_items < 4) {
-      fit$col_widths <- 1
-      return(fit)
-    }
-
-    if (n_items <= 7) {
-      # sizes 4-7 are special cased to use (2 * items) columns
-      fit$col_widths <- if (prefer_wider) n_items else 2
-      return(fit)
-    }
-  }
-
-  # We're in fixed 12-column mode, manually pick for small numbers where the
-  # algorithm would otherwise pick a suboptimal column widths.
-  if (fit$n_cols == 12) {
-    if (n_items <= 3) {
-      fit$col_widths <- c(12, 6, 4)[n_items]
-      return(fit)
-    }
-
-    # n == 4 gives 6 for wide (2x2) and 3 for narrow columns (1x4)
-
-    if (n_items == 5 || n_items == 7) {
-      fit$col_widths <- if (prefer_wider) 4 else 3
-      return(fit)
-    }
-
-    if (n_items == 6) {
-      # (2x3) for wide and (1x6) for narrow items
-      fit$col_widths <- if (prefer_wider) 4 else 2
-      return(fit)
-    }
-  }
-
-  fctrs <- if (prefer_wider) c(6, 4, 3) else c(2, 3, 4)
-
-  col_units <- n_items * fctrs
-  rows <- ceiling(col_units / 12)
-  total_units <- rows * 12
-  empty_units <- total_units - col_units
-
-  fit$col_widths <- fctrs[which.min(empty_units)]
-
-  fit
-}
-
-
-
 grid_item_container <- function(el, ..., fillable = TRUE) {
   div(
     ...,
-    class = "bslib-gap-spacing",
+    class = "bslib-grid-item bslib-gap-spacing",
```

Comment:
I think I'd be in support for dropping this if aren't adding rules/logic for it?

```suggestion
    class = "bslib-gap-spacing",
```

Reply:
I'm okay with taking it out, but I do like that it documents the object and gives the user a hook. Yes, they could `.bslib-grid > *` but as a user this always feels brittle, so I have a slight preference for including a stable selector even if we don't use it directly.